File: data/gpt-4o/readable_format_HPEC/cpp/file175.cpp
#include "ft_data.h"
#include <complex>

void compute_initial_conditions(std::complex<double> u0[], int d1, int d2, int d3) {
    int k, j;
    double x0, start, an, dummy;
    double starts[nz];

    start = seed;
    ipow46(a, 0, &an);
    dummy = randlc(&start, an);
    ipow46(a, 2*nx*ny, &an);

    starts[0] = start;
    for (k = 1; k < dims[2]; k++) {
        dummy = randlc(&start, an);
        starts[k] = start;
    }

    for (k = 0; k < dims[2]; k++) {
        x0 = starts[k];
        for (j = 0; j < dims[1]; j++) {
            vranlc(2*nx, &x0, a, &u0[j + d1*(k + d2*j)]);
        }
    }
}

File: data/CodeLlama/readable_format_HPEC/fortran/file175.f90
subroutine compute_initial_conditions(u0, d1, d2, d3)



use ft_data
implicit none

integer d1, d2, d3
double complex u0(d1+1, d2, d3)
integer k, j
double precision x0, start, an, dummy, starts(nz)


start = seed
call ipow46(a, 0, an)
dummy = randlc(start, an)
call ipow46(a, 2*nx*ny, an)

starts(1) = start
do k = 2, dims(3)
dummy = randlc(start, an)
starts(k) = start
end do

do k = 1, dims(3)
x0 = starts(k)
do j = 1, dims(2)
call vranlc(2*nx, x0, a, u0(1, j, k))
end do
end do

return
end
